src/HassClient/Client/HassClient.cs

```diff
@@ -319,6 +357,17 @@ public class HassClient : IHassClient
             _pipelineFactory = pipelineFactory;
             _wsFactory = wsFactory;
             _httpClient = httpClient;
+
+            HassEventsObservable = Observable.FromEventPattern<EventHandler<HassEvent>, HassEvent>(

```

Comment:
There is a Class Subject<T> that does the same as the events but is a lot simpler It implements Both IObeservable AND IObserver. You can make a private field
```
private Subject<HassEvent> _hassEventSubject = new();

public IObservable HassEventsObservable => _hassEventSubject;
```
And then when you want to publish an event call `_hassEventSubject.OnNext(new HassEvent(...));`